eb_hooks.py

```diff
@@ -656,6 +714,26 @@ def inject_gpu_property(ec):
                 ec[key] = '\n'.join([ec_dict[key], value])
         else:
             ec[key] = value
+
+    # Check if cuDNN is in the dependencies, if so add the 'gpu' Lmod property
+    if ('cuDNN' in [dep[0] for dep in iter(ec_dict['dependencies'])]):
+        ec.log.info("Injecting gpu as Lmod arch property and envvar with cuDNN version")
+        key = 'modluafooter'
+        value = 'add_property("arch","gpu")'
+        cudnn_version = 0
+        for dep in iter(ec_dict['dependencies']):
+            # Make cuDNN a build dependency only (rpathing saves us from link errors)
+            if 'cuDNN' in dep[0]:
+                cudnn_version = dep[1]
+                ec_dict['dependencies'].remove(dep)
+                if dep not in ec_dict['builddependencies']:
+                    ec_dict['builddependencies'].append(dep)
+        value = '\n'.join([value, 'setenv("EESSICUDNNVERSION","%s")' % cudnn_version])
+        if key in ec_dict:
+            if not value in ec_dict[key]:
```

Comment:
This check is probably no longer good enough, we're looking for the exact string, but that is not likely to exist (even though the `add_property("arch","gpu")` most likely does exist since the applications also should have a `CUDA` dep). What we really need to do is

- Grab what is there already
- Split it on `\n`
- Add any missing elements
- Put it back together again and replace it

Either this, or only the modify/add the `modluafooter` once in the entire function

Reply:
No, because the value is a composite string of `property` and the `setenv`, and the property will already exist from the CUDA part of this hook

Reply:
The module file for `cuDNN` contains the following
```lua
-- Built with EasyBuild version 4.9.1

add_property("arch","gpu")
setenv("EESSICUDAVERSION","12.1.1")
```

For something that builds on top of `cuDNN`, we would the above and something like
```lua
setenv("EESSICUDNNVERSION","8.9.2.26")
```

Reply:
As currently implemented, for something that builds on top of cuDNN I believe you will have 
```
-- Built with EasyBuild version 4.9.1

add_property("arch","gpu")
setenv("EESSICUDAVERSION","12.1.1")
add_property("arch","gpu")
setenv("EESSICUDNNVERSION","8.9.2.26")
```

Reply:
Ack. Working on something to implement the desired footer (and avoiding duplication of code).

Reply:
Updated the function. It still produces the same footer for `cuDNN`. I guess a real test would be a build that uses `cuDNN`. @ocaisa can you check if the function looks better now?